scripts/EndOfRunMonitor/ISISendOfRunMonitor.py

```diff
@@ -0,0 +1,111 @@
+
+
+def get_data_and_check(last_run_file):
+    """ Gets the data from the last run file and checks it's format """
+    data = last_run_file.readline().split()
+    if len(data) != 3:
+        raise Exception("Unexpected last run file format")
+    return data
+
+
+class InstrumentMonitor(threading.Thread):
+    def __init__(self, instrument_name, client, lock):
+        super(InstrumentMonitor, self).__init__()
+        self.client = client
+        self.instrumentName = instrument_name
+        if DEBUG:
+            self.instrumentFolder = '.'
+        else:
+            self.instrumentFolder = INST_FOLDER % self.instrumentName
+        self.instrumentSummaryLoc = self.instrumentFolder + SUMMARY_LOC
+        self.instrumentLastRunLoc = self.instrumentFolder + LAST_RUN_LOC
+        self.instrumentDataFolderLoc = self.instrumentFolder + DATA_LOC
+        self.lock = lock
+        
+    def build_dict(self, last_run_data):
+        """ Uses information from lastRun file, 
+        and last line of the summary text file to build the query 
+        """
+        filename = ''.join(last_run_data[0:2])  # so MER111 etc
+        run_data_loc = self.instrumentDataFolderLoc + filename + get_file_extension()
+        return {
+            "rb_number": self._get_RB_num(),
+            "instrument": self.instrumentName,
+            "data": run_data_loc,
+            "run_number": last_run_data[1],
+            "facility": "ISIS"
+        }
+
+    def _get_RB_num(self):
+        """ Reads last line of summary.txt file and returns the RB number
+        """
+        summary = self.instrumentSummaryLoc
+        with open(summary, 'rb') as st:
+            last_line = st.readlines()[-1]
+            return last_line.split()[-1]
+
+    def run(self):
+        """ Works to actually monitor the last run file
+        """
+        try:
+            with open(self.instrumentLastRunLoc) as lr:
+                data = get_data_and_check(lr)
+                last_run = data[1]
+
+            while True:  # send thread to sleep, use Timer objects
+                time.sleep(TIME_CONSTANT)
+                with open(self.instrumentLastRunLoc) as lr:
+                    data = get_data_and_check(lr)
+                    if (data[1] != last_run) and (int(data[2]) == 0):
+                        last_run = data[1]
+                        self.send_message(data)
+
+        except Exception as e:
+            logging.exception("Error on loading file: ")
+            raise e
+
+    def send_message(self, last_run_data):
+        """Puts message together and sends it, along with logging
+        """
+        with self.lock:
+            data_dict = self.build_dict(last_run_data)
+        if not DEBUG:
+            self.client.send('/queue/DataReady', json.dumps(data_dict))
+        logging.info("Data sent: " + str(data_dict))
+
+
+def main():
+    activemq_client = StompClient([("autoreduce.isis.cclrc.ac.uk", 61613)], 'autoreduce', '1^G8r2b$(6', 'RUN_BACKLOG')
```

Comment:
Please don't store passwords to git. In this case autoreduce.isis.cclrc.ac.uk is not visible outside the isis, but otherwise..... Please replace 1^G8r2b$(6 with a string like 'xxxxxxxxxx' and for good measure setup a new password with ActiceMQ